packages/x-components/src/x-modules/scroll/store/types.ts

```diff
@@ -0,0 +1,159 @@
+import { ScrollDirection } from '../../../components/scroll/scroll.types';
+import { XStoreModule } from '../../../store';
+import { Dictionary } from '../../../utils/types';
+
+/**
+ * Scroll store state.
+ *
+ * @public
+ */
+export interface ScrollState {
+  /**
+   * State data of the scroll components.
+   */
+  data: Dictionary<ScrollComponentState>;
+  /**
+   * The `[data-scroll]` value of the pending to restore scroll.
+   */
+  pendingScrollTo: string;
+}
+
+/**
+ * Contains all the state of a scroll component.
+ *
+ * @public
+ */
+export interface ScrollComponentState {
+  /**
+   * The position in pixels the user has scrolled down.
+   */
+  position: number;
```

Comment:
Maybe `positionInPx`?

Reply:
I don't think we need to be that specific here, but if @LuisMartinez15 also tells me to add it I won't fight you both. In any case, it is `positionPx`, no `positionInPx`